src/vm/functions/boolean.rs

```diff
@@ -38,8 +43,7 @@ pub fn special_and(args: &[SymbolicExpression], env: &mut Environment, context:
     Ok(Value::Bool(true))
 }
 
-pub fn native_not(args: &[Value]) -> Result<Value> {
-    check_argument_count(1, args)?;
-    let value = type_force_bool(&args[0])?;
+pub fn native_not(input: Value) -> Result<Value> {
+    let value = type_force_bool(&input)?;
```

Comment:
I think that we're missing
```
runtime_cost!(cost_functions::NOT, env, args.len())?;
```
?

Reply:
For the functions that are implemented with `NativeFunction`, they include their cost_function in the functions enum. See src/vm/functions/mod.rs line 112 and src/vm/mod.rs line 125.

Those simple native functions have some other benefits too -- the enum handles checking the argument length, eval'ing the argument, and passing it in as an owned reference.